src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Data/BranchStrategy.cs

```diff
@@ -0,0 +1,137 @@
+			 *
+			 * For instance if we have
+			 * List (a.k.a. 'root', if enumerated we get: Item 1, Item 2, Item 3, Item 4)
+			 *		Group 1
+			 *			Item 1
+			 *			Item 2
+			 *		Group 2
+			 *			Item 3
+			 *			Item 4
+			 *
+			 *	Then if we replace the items of group 2 (i.e. index 1), using a 'Reset'
+			 *		Group 2
+			 *			Item A
+			 *			Item B
+			 *
+			 * The window's collection (windows 10.15063) view will raise
+			 *		root: Reset @ 1
+			 *		group[1]: Reset @ 0
+			 *
+			 * The issue is that the first event makes **absoluteley no sense** (the index is the index of the group while on root we usually deals with item indexes).
+			 *
+			 * As currently we will do either some add / remove in groups, either a **FULL** reset on the whole grouped collection,
+			 * we admit this as limitiation and we won't support this.
+			 *
+			 * Behavior of full reset in grouped collection is validated by the test 'Given_BindableCollection.When_Grouped_SwitchWithReset'
+			 *
+			 */
+
+			var view = default(ICollectionView);
+
+			// Extended property bag that can be updated by all facets to expose some extended info of their internal for binding
+			// (e.g. Pagination.IsLoadingMoreItems)
+			var extendedPropertiesFacet = new BindableCollectionExtendedProperties();
+
+			// Init the flat tracking view (ie. the flatten view of the groups that can be consumed directly by the ICollectionView properties)
+			var flatCollectionChanged = new FlatCollectionChangedFacet(() => view ?? throw new InvalidOperationException("The owner provider must be resolved lazily!"));
+			var flatSelectionFacet = new SelectionFacet(() => view ?? throw new InvalidOperationException("The owner provider must be resolved lazily!"));
+			var flatPaginationFacet = new PaginationFacet(source, extendedPropertiesFacet);
+
+			// Init the groups tracking
+			var groupsChanged = new CollectionChangedFacet(() => view?.CollectionGroups ?? throw new InvalidOperationException("The owner provider must be resolved lazily!"));
+			var groupsCollection = new CollectionFacet(groupsChanged, convertResetToClearAndAdd: ObservableCollectionKind.Vector, onReseted: flatCollectionChanged.NotifyReset);
+			var groupsMappingFacet = new DataLayerCollection();
+
+			// Init the views
+			var groupsView = new MapView(
+				groupsCollection,
+				groupsChanged,
+				groupsMappingFacet.ToConverter()
+#if NETFX_CORE
+				, enableWinRTOutOfRangeHack: true
+#endif
+			);
+			var flatView = new FlatView(
+				source: groupsCollection,
+				collectionChangedFacet: flatCollectionChanged,
+				selectionFacet: flatSelectionFacet,
+				paginationFacet: flatPaginationFacet,
+				groups: groupsView);
+
+			// Init the view variable in order to make it accessible to facets
+			view = flatView;
+
+			return (groupsCollection, view, new object[] { flatCollectionChanged, flatPaginationFacet, flatSelectionFacet, groupsMappingFacet, extendedPropertiesFacet });
```

Comment:
Is this something we do often? Could be use an ArrayPool?

Reply:
Not really, it could be interesting indeed, but as we are transferring ownership of that array, it could be misleading.